src/Compilers/CSharp/Portable/Emitter/Model/MethodSymbolAdapter.cs

```diff
@@ -82,6 +82,16 @@ Cci.ITypeReference Cci.ITypeMemberReference.GetContainingType(EmitContext contex
             }
 
             NamedTypeSymbol containingType = AdaptedMethodSymbol.ContainingType;
+
+            if (AdaptedMethodSymbol is SynthesizedExtensionMarker)
+            {
+                return ((SourceMemberContainerTypeSymbol)containingType.ContainingType).GetExtensionGroupingInfo().GetCorrespondingMarkerType((SourceNamedTypeSymbol)containingType);
+            }
+            else if (containingType.IsExtension)
```

Comment:
>containingType.IsExtension [](http://example.com/codeflow?start=21&length=26)

nit: consider `AdaptedMethodSymbol.GetIsNewExtensionMember()` to be consistent with other adapter checks (I know that we already have `containingType` at hand above, but the consistency improves readability somewhat)